crates/cli/src/report.rs

```diff
@@ -0,0 +1,522 @@
+use std::{
+    cell::Cell,
+    collections::HashMap,
+    fs::File,
+    hash::{Hash, Hasher},
+    io::{BufRead, BufReader, Write},
+    path::{Path, PathBuf},
+};
+
+use serde::Serialize;
+use sightglass_analysis::{effect_size, report_stats::BenchmarkStats};
+use sightglass_data::{EffectSize, Format, Measurement, Phase};
+use structopt::StructOpt;
+use vega_lite_4::{
+    AxisBuilder, ColorClassBuilder, EdEncodingBuilder, LegendBuilder, Mark, NormalizedSpecBuilder,
+    XClassBuilder, YClassBuilder,
+};
+
+const TEMPLATE: &'static str =
+    include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/src/report.jinja"));
+
+/// Generate an HTML report for a given set of raw inputs
+#[derive(Debug, StructOpt)]
+#[structopt(name = "report")]
+pub struct ReportCommand {
+    /// The format of the input data. Either 'json' or 'csv'; if not provided
+    /// then we will attempt to infer it from provided filenames (default: json).
+    #[structopt(short = "i", long = "input-format")]
+    input_format: Option<Format>,
+
+    /// Output HTML file path
+    #[structopt(short = "o", long = "output-file", default_value = "report.html")]
+    output_path: PathBuf,
+
+    /// Name of the baseline to use; if not provided, the first engine encountered
+    /// in the ordered input files will be used.
+    #[structopt(short = "b", long = "baseline-engine")]
+    baseline_engine: Option<String>,
+
+    /// Significance level for statistical tests (default: 0.05 for 95% confidence)
+    #[structopt(long = "significance-level", default_value = "0.05")]
+    significance_level: f64,
```

Comment:
Ensure this is plumbed and used consistently throughout all the report code.  Elsewhere it seems like it was being ignored.